integration-test/src/main/java/com/sequenceiq/it/cloudbreak/testcase/e2e/l0promotion/InternalSdxSshAndCmAccessTest.java

```diff
@@ -0,0 +1,144 @@
+import com.sequenceiq.it.cloudbreak.dto.stack.StackTestDto;
+import com.sequenceiq.it.cloudbreak.dto.ums.UmsTestDto;
+import com.sequenceiq.it.cloudbreak.testcase.e2e.sdx.PreconditionSdxE2ETest;
+import com.sequenceiq.it.cloudbreak.util.clouderamanager.ClouderaManagerUtil;
+import com.sequenceiq.it.cloudbreak.util.ssh.SshJUtil;
+import com.sequenceiq.it.util.ResourceUtil;
+import com.sequenceiq.sdx.api.model.SdxClusterStatusResponse;
+
+public class InternalSdxSshAndCmAccessTest extends PreconditionSdxE2ETest {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(SetWorkloadPasswordAction.class);
+
+    @Inject
+    private SdxTestClient sdxTestClient;
+
+    @Inject
+    private RecipeTestClient recipeTestClient;
+
+    @Inject
+    private FreeIpaTestClient freeIpaTestClient;
+
+    @Inject
+    private UmsTestClient umsTestClient;
+
+    @Inject
+    private SshJUtil sshJUtil;
+
+    @Inject
+    private ClouderaManagerUtil clouderaManagerUtil;
+
+    @Test(dataProvider = TEST_CONTEXT)
+    @Description(
+            given = "there is a running Manowar SDX cluster in available state",
+            when = "SSH to the SDX nodes with 'admin' then acting user with new workload password, where the 'post-install' recipe had been installed.",
+            and = "log in to the Cloudera Manager, where 'knox-IDBROKER-BASE' role config mappings then related Cloudera Manager user details have been read",
+            then = "SSH and Cloudera Manager API access should be successful, 'post-install' recipe files and requested Cloudera Manager details should be" +
+                    " present"
+    )
+    public void testSdxSshAndCmAccessWithNewWorkloadPassword(TestContext testContext) throws IOException {
+        String freeIpa = testContext.given(FreeIpaTestDto.class).when(freeIpaTestClient.describe()).getResponse().getName();
+        String sdxInternal = resourcePropertyProvider().getName();
+        String cluster = resourcePropertyProvider().getName();
+        String clouderaManager = resourcePropertyProvider().getName();
+        String recipeName = resourcePropertyProvider().getName();
+        String stack = resourcePropertyProvider().getName();
+        String filePath = "/post-install";
+        String fileName = "post-install";
+        String masterInstanceGroup = "master";
+        String idbrokerInstanceGroup = "idbroker";
+        String newWorkloadPassword = "Admin123@";
+        String workloadUsername = getWorkloadUsername(testContext);
+
+        testContext
+                .given(clouderaManager, ClouderaManagerTestDto.class)
+                .given(cluster, ClusterTestDto.class).withBlueprintName(getDefaultSDXBlueprintName()).withValidateBlueprint(Boolean.FALSE)
+                .withClouderaManager(clouderaManager)
+                .given(RecipeTestDto.class).withName(recipeName).withContent(generateRecipeContent())
+                .withRecipeType(POST_CLOUDERA_MANAGER_START)
```

Comment:
why do we need a recipe?

Reply:
We do not have E2E recipe test. So I think this will be useful as a side effect for SSH test. Beyond this I think a proper flow if we not just briefly SSH to the nodes even more checking something there (a newly created file is present).

Reply:
if we need a recipe e2e test, i would create separate one for it and not mix different test cases. From authorization point of view, just SSHing is enough, checking files is not needed in these test cases I think.

Reply:
by default every user can read log files, etc, on nodes, there is no specific authorization logic for files, so I would not bring file permission into these test cases.